src/python/analyzer_executor/src/analyzer_executor_lib/analyzer_executor.py

```diff
@@ -54,157 +53,331 @@ def get(self, key):
+                    )
+                    emit_event(s3, result, self.is_local)
+                    self.update_msg_cache(
+                        analyzer, result.root_node_key, message["key"]
+                    )
+                    self.update_hit_cache(analyzer_name, result.root_node_key)
+
+                assert not isinstance(
+                    result, ExecutionFailed
+                ), f"Analyzer {analyzer_name} failed."
+
+            p.join()
+
+    def exec_analyzers(
+        self,
+        dg_client,
+        file: str,
+        msg_id: str,
+        nodes: List[BaseView],
+        analyzers: Dict[str, Analyzer],
+        sender: Any,
+    ):
+        if not analyzers:
+            self.logger.warning("Received empty dict of analyzers")
+            return
 
-            queries = analyzer.get_queries()
-            if isinstance(queries, list) or isinstance(queries, tuple):
-                querymap[an_name].extend(queries)
-            else:
-                querymap[an_name].append(queries)
+        if not nodes:
+            self.logger.warning("Received empty array of nodes")
 
-        for an_name, queries in querymap.items():
-            analyzer = analyzers[an_name]
+        for node in nodes:
+            querymap: Dict[str, List[Queryable]] = defaultdict(list)
 
-            for query in queries:
-                response = query.query_first(dg_client, contains_node_key=node.node_key)
-                if response:
-                    LOGGER.debug(
-                        f"Analyzer '{an_name}' received a hit, executing on_response()"
+            for an_name, analyzer in analyzers.items():
+                if self.check_caches(file, msg_id, node.node_key, an_name):
+                    continue
+
+                queries = analyzer.get_queries()
+                if isinstance(queries, list) or isinstance(queries, tuple):
+                    querymap[an_name].extend(queries)
+                else:
+                    querymap[an_name].append(queries)
+
+            for an_name, queries in querymap.items():
+                analyzer = analyzers[an_name]
+
+                for query in queries:
+                    response = query.query_first(
+                        dg_client, contains_node_key=node.node_key
                     )
-                    analyzer.on_response(response, sender)
+                    if response:
+                        self.logger.debug(
+                            f"Analyzer '{an_name}' received a hit, executing on_response()"
+                        )
+                        analyzer.on_response(response, sender)
 
+    def execute_file(
+        self, name: str, file: str, graph: SubgraphView, sender, msg_id, chunk_size
+    ):
+        try:
+            pool = ThreadPool(processes=4)
 
-def chunker(seq, size):
-    return [seq[pos : pos + size] for pos in range(0, len(seq), size)]
+            exec(file, globals())
+            client = GraphClient()
```

Comment:
seems like we could just make a `self.client = GraphClient()` in the constructor`

Reply:
since theres at least one other place that constructs a graphclient

Reply:
so I tried this and started getting environment errors from GraphClient, I think the environment changes after we start executing within the lambda, so initializing before that is a bad time.  I'll leave this alone for now